### ISSUE TITLE
[Feature] locator.children() returning all children of given selector.

### ISSUE THREAD

**Issue by andrzejsokolowski** [microsoft/playwright#17703]
Honestly such a fundamental feature I can't believe it's not here yet.
Cypress has it, it's useful (and easy to add).
https://docs.cypress.io/api/commands/children

**Comment by aslushnikov** [comment]
@andrzejsokolowski In the modern web, adding intermediate divs in the components happens all the time. Each change like that will be destroying the tests that use `locator.children()`.

What's your usecase? Why would you need `locator.children()`?

**Comment by andrzejsokolowski** [comment]
I believe i had this problem when trying to select an empty text element. Since locator objects can only get elements, i had this structure:
`<div>
    <span> textOne </span>
     textTwo
  </div>`
  
 And I didn't find a way to get this one empty textTwo without also getting textOne with .textContent()
 In the end i settled for a
`(await this.infoCard('total-allocation').textContent()).slice(16)` to eliminate textOne.

with `locator.children()` i could easily use `.nth(1)` to get that element

**Comment by yury-s** [comment]
You can access corresponding text node using DOM API in an eval, something like this:

```js
console.log(await page.locator('div').evaluate(div => div.childNodes[2].textContent));
```

Closing this issue as we are not planning to support locator.parent()/.children() methods.

**Comment by MessiasMorten** [comment]
Sorry for opening this up again, but if there is no intent to support parent and children methods for locators it would be significantly more difficult to be able to scope in on components within third party packages like Angular Material, Mobiscroll etc, where you as the developer have no control over the identifiers.

**Comment by kyle-uc** [comment]
Ok, here's my use case: I have two elements whose location within the page can be changed. I want to assert that one of them appears earlier in the DOM than the other. My approach would be to select the parent element and iterate through its children until one of the two selectors was matched. In practice, my tests are a controlled environment and I _can_ predict the `n` (as in `nth-child`), but its not very robust. I don't want my test to fail because other elements moved around. I just want it to assert that element A comes before element B

**Comment by j2yes** [comment]
In my case, below code was useful.

`page.locator("parent-selector").locator("*").all()`

**Comment by vr-patel** [comment]
It would be beneficial if the playwright team offers assistance for methods related to get the parent-child nodes.

use case, where it is necessary to retrieve child nodes.

I am managing a webpage with numerous tabs, where some tabs are displayed based on the browser width while others are hidden and can only be accessed through a dropdown menu with a "more" button. The number of visible tabs varies depending on the browser width. I intended to obtain a compilation of both visible and concealed tabs, along with additional details.

This is what the HTML content appears to be.

`<ul class="tabs">
      <li>Tab 1</li>
      <li>Tab 2</li>
      <li>Tab 3</li>
      <li>Tab 4</li>
      <li>Tab 5</li>
      <li class="dropdown">
        More
        <ul>
          <li>Tab 6</li>
          <li>Tab 7</li>
          <li>Tab 8</li>
          <li>Tab 9</li>
          <li>Tab 10</li>
        </ul>
      </li>
    </ul>`

In this HTML document, I aimed to retrieve the list of visible tabs. However, when attempting to do so with the line below, I obtained all the `<li>` tags, including the Dropdown.
